src/TMainConsole.h

```diff
@@ -103,6 +103,8 @@ class TMainConsole : public TConsole
     bool loadMap(const QString&);
     bool importMap(const QString&, QString* errMsg = Q_NULLPTR);
 
+    static const char* scmProperty_HostName;
```

Comment:
Because you would still need to use them to embed the profile name into the base (`QObject`) class **unless** the thing returned from `mpHBoxLayout_profileContainer->itemAt(profileIndex)->widget()`, which seems to be a `QWidget` and NOT the `TConsole` or `TMainConsole` derived class can be cast to the derived class with `dynamic_cast<TMainConsole*>()` as https://stackoverflow.com/q/18873871/4805858 suggests - but I'm not sure about the requirement for there to be at least one `virtual function`.

I do not need a `TMainConsole*` here - I just need to know which `QLayoutItem*` points to the top level *container* for which profile - so embedding that information in the base `QObject` that each item is derived from works fine for this.

Reply:
Yes it does, but how can get get rid of unintuitive things like `scmProperty_HostName`? Do we really need it? I don't think so because it's set in 1 place and read in 1 other.

Reply:
Well either we declare it in an obvious place (at the top of the compilation unit) - or we have a raw literal string `"HostName"` (though not one that is converted to a `QString`!) used in two places where they must be the same.

As it happens I have already used `scmProperty_` as a prefix on some constants elsewhere (see `TTimer::scmProperty_HostName` and `TTimer::scmProperty_TTimerId`) for the same sort of thing {adding meta-data to a `QObject` derived class} and the meta-properties that are involved are accessed in the `mudlet` and `TimerUnit` classes as well as the `TTimer` one. To be fair it would be nicer not to have to tie the definitions to a particular class - in this case the `TMainConsole` one but it does, I suppose say where to go to look for the raw string being used - even though the whole idea of using a ***constant*** instead of the same literal in different places, **even** though it is only **two** for this one, is to abstract that away. Of course `TTimer::scmProperty_HostName` and `TMainConsole::scmProperty_HostName` also have the same **value** but they are used in different places to store the same piece of data in different `QObject`-derived class instances: `TTimer` and `TMainConsole` ones respectively.

tl;dr; Are you insisting that I replace the uses of `TMainConsole::scmProperty_HostName` with `"HostName"`?

Reply:
Yeah. Abstractions, just like any tool, are best used sparingly and in this case it's adding more trouble than help. Something like `TMainConsole` vs `TConsole` - good idea, makes sense - but this is too little and the price is too much.

Reply:
Done in https://github.com/Mudlet/Mudlet/pull/4976/commits/6405a56e8c119b57366cb14d3aadced82a87646f.